### DISCUSSION TITLE
raw socket load testing

### DISCUSSION THREAD

**Discussion by fschaeffler** [artilleryio/artillery#980] [Artillery]
One of our application servers is accepting requests via raw sockets. Those are **not WebSockets**. I searched the documentation but can’t really find out anything about raw sockets. We want to do a stress test on that particular server, to find out about our actual limits.

My question is now, are raw sockets actually supported by Artillery?

If raw sockets aren’t supported, is there any possibility to implement this via an e.g. plugin?

**Example of raw sockets in Node.js**
```
const net = require('net')

const server = net.createServer(socket => {
  socket.on('data', data => {
    console.log('some data received', data)

    const response = 'some response data'
    socket.write(response)
  })
})

server.listen()
```

**Comment by hassy** [comment]
hi @fschaeffler👋 yep, that can be done via a plugin. There's an existing one which may work:

https://github.com/limiter121/artillery-engine-tcp

and if not, it could be a starting point for your own. Happy to help with that.
